### PR TITLE
Pathing module addition

### PR DESCRIPTION
**Pull request by D13Lee**
importable, better conventions and organisation, brief explanation of usage at the top of pather.js

### CODE REVIEW

**Comment by D13Lee** [comment]
Updated main.js, forgot to add new tests in previous commit.
added a spawner class, with tests included.

File: main.js [hunk 1]
```diff
@@ -3,4 +3,6 @@ require('babel-register')({
     presets: ['env']
 });
 
-module.exports = require('./test.js')
\ No newline at end of file
+module.exports = require('./test.js')
+module.exports = require('./patherTests.js')
+module.exports = require('./mobSpawnerTests.js')
```
Comment (Squishy123) [inline main.js]: Might want to look into using a .babelrc config file.
You can use babel-node for development and then using a babel script for building. 
Take a look at [engine repo](https://github.com/Project-Rekt/engine) at the .babelrc and package.json scripts.

File: mobSpawnerTests.js [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+    }
+
+    spawnerTest(){
+        var results = []
+        results.push(this.spawnerTest1())
+        results.push(this.spawnerTest2())
+        results.push(this.spawnerTest3())
+        results.push(this.spawnerTest4())
+        results.push(this.spawnerTest5())
+        results.push(this.spawnerTest6())
```
Comment (Squishy123) [inline mobSpawnerTests.js]: You can just instantiate the array with these values:
e.g. 
```
let results=[this.spawnerTest1(), this.spawnerTest2(), ...]
```

File: mobSpawnerTests.js [hunk 3]
```diff
@@ -0,0 +1,92 @@
+/*
+ * tests for the spawner object
+ */
+
+import { Spawner } from './src/classes/spawner';
+class spawnerTester{
```
Comment (Squishy123) [inline mobSpawnerTests.js]: Classnames should be instantiated
```suggestion
class SpawnerTester {
```

File: main.js [hunk 4]
```diff
@@ -3,4 +3,6 @@ require('babel-register')({
     presets: ['env']
 });
```
Comment (Squishy123) [inline main.js]: You might want to move this into a .babelrc config:
Use babel-node for development and babel -d for builds: 
Take a look at engine repo .babelrc and package.json https://github.com/Project-Rekt/engine

File: main.js (hunk 1 above)
Comment (Squishy123) [inline main.js]: You can probably use a test runner like jest https://jestjs.io/

File: mobSpawnerTests.js [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+ */
+
+import { Spawner } from './src/classes/spawner';
+class spawnerTester{
+    constructor(){
+
+    }
+
+    spawnerTest(){
+        var results = []
```
Comment (Squishy123) [inline mobSpawnerTests.js]: Stay away from var, use let instead 
```suggestion
let results = []
```

**Review by Squishy123 — DISMISSED** [review]
Just did a quick readthrough, in general 
- try changing exports to default if theres only one in the file
- change all var to let
- maybe look into jest for tests

File: main.js (hunk 4 above)
Comment (D13Lee) [reply main.js]: @KingJeremyNg I think this one is for you

**Comment by D13Lee** [comment]
Deleted old main.js addition, because of automatic directory import Jeremy added

**Comment by D13Lee** [comment]
won't add anything major until after this merge because there is too much not approved yet haha

**Comment by D13Lee** [comment]
no more reviews have been submitted, assumed to be accepted :)

**Pull request merged**